lib/components-schema-v1_0_x.ts

```diff
@@ -5,6 +5,302 @@
  */
 
  /* tslint:disable:max-line-length variable-name */
+
+const componentPropertyDefinition = {
+    name: {
+        type: 'string',
+        description: 'Unique identifier of component property',
+        minLength: 3,
+    },
+    label: { type: 'string', description: 'Display label of Component property' },
+    directiveKey: { type: 'string', description: 'Directive key for properties that use a directive data type' },
```

Comment:
It looks differently that an example which you showed in slack. If I understand correctly the whole property configuration should be described if directiveKey is set? For example I have two components, both have image directive but directives key are different. Now how to apply image-editor property to them?

Reply:
Iteresting that changes in boilerplate project is the same as an example in slack. So why not to describe all three posible variants of property item of properties field? Now there are two only and the second one kind of covers anonymous property and reference property with directiveKey.

Reply:
Not sure what you mean, but the entry from componentProperties is merged with the object from the properties array of a component property. The directiveKey can be specified in the properties list of the component.

Reply:
I mean that now it is possible to add non valid property, for example without name and directiveKey, or other variant. Now the schema does not guarantee that the item added to 'properties' property of component is valid

Reply:
Name is not required.

I will see if I can add validation after you merge your task.